psl/psl-core/src/builtin_connectors/capabilities_support.rs

```diff
@@ -0,0 +1,73 @@
+use crate::datamodel_connector::{Connector, ConnectorCapabilities, ConnectorCapability};
+use cfg_if::cfg_if;
+
+cfg_if! {
+    if #[cfg(all(feature="mysql", not(any(feature = "postgresql", feature="sqlite", feature = "cockroachdb", feature="mssql", feature="mongodb"))))] {
+        #[inline(always)]
+        const fn can_have_capability_impl(capability: ConnectorCapability) -> bool {
+            check_comptime_capability(super::mysql_datamodel_connector::CAPABILITIES, capability)
+        }
+
+        pub fn has_capability(_: &dyn Connector, capability: ConnectorCapability) -> bool {
+            can_have_capability_impl(capability)
+        }
+    } else if #[cfg(all(feature="sqlite", not(any(feature = "postgresql", feature="mysql", feature = "cockroachdb", feature="mssql", feature="mongodb"))))] {
+        #[inline(always)]
+        const fn can_have_capability_impl(capability: ConnectorCapability) -> bool {
+            check_comptime_capability(super::sqlite_datamodel_connector::CAPABILITIES, capability)
+        }
+
+        #[inline(always)]
+        pub fn has_capability(_: &dyn Connector, capability: ConnectorCapability) -> bool {
+            can_have_capability_impl(capability)
+        }
+    } else if #[cfg(all(feature="postgresql", not(any(feature = "sqlite", feature="mysql", feature = "cockroachdb", feature="mssql", feature="mongodb"))))] {
+        #[inline(always)]
+        const fn can_have_capability_impl(capability: ConnectorCapability) -> bool {
+            check_comptime_capability(super::postgres_datamodel_connector::CAPABILITIES, capability)
+        }
+
+        #[inline(always)]
+        pub fn has_capability(_: &dyn Connector, capability: ConnectorCapability) -> bool {
+            can_have_capability_impl(capability)
+        }
+    } else {
+        #[inline(always)]
+        const fn can_have_capability_impl(_: ConnectorCapability) -> bool {
+            true
+        }
+
+        #[inline(always)]
+        pub fn has_capability(connector: &dyn Connector, capability: ConnectorCapability) -> bool {
+            connector.capabilities().contains(capability)
+        }
+    }
+}
+
+/// Helper function for determining if engine, compiled with the current settings
+/// can potentially have provided capability on. Useful for single-connector builds and can
+/// be used to exclude certain code that we know for sure can't be executed for current connector.
+/// Has no effect on multi-connector builds
+/// # Example
+/// ```ignore
+/// if !can_have_capability(ConnectorCapability::FullTextSearch) {
+///    unreachable!()
+/// }
+/// ... // if compiled for a single connector, optimizer will exclude the following code if connector does not support full text search
+/// ```
+#[inline(always)]
+pub const fn can_have_capability(cap: ConnectorCapability) -> bool {
+    can_have_capability_impl(cap)
+}
```

Comment:
Naming suggestion:

```suggestion
pub const fn allow_capability_check(cap: ConnectorCapability) -> bool {
    allow_capability_check_impl(cap)
}
```

Reply:
I honestly don't get the suggested name. Any build allowed to have any capability check at any time, this PR does not change this.
What `can_have_capability` does is tells if it is possible that with currently built features, any connector in the build would potentially return `true` for given capability.

Reply:
Basically, if `can_have_capability(cap)` is `false`, then none of `connector.has_capability(cap)` will return `true` for current build configuration. 
If `can_have_capability(cap)` is `true`, then the result of `connector.has_capability(cap)` can be either `true` or `false`. 
If there is only one connector configured, `can_have_capability(cap)` and `connector.has_capability(cap)` will always be the same.